shared/src/store/action-reducers/alarm-group.ts

```diff
@@ -0,0 +1,163 @@
+    public readonly alarmGroupVehicleId!: UUID;
+
+    @IsNumber()
+    public readonly time!: number;
+}
+export class RemoveAlarmGroupVehicleAction implements Action {
+    @IsString()
+    public readonly type = '[AlarmGroup] Remove AlarmGroupVehicle';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly alarmGroupId!: UUID;
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly alarmGroupVehicleId!: UUID;
+}
+
+export namespace AlarmGroupActionReducers {
+    export const addAlarmGroup: ActionReducer<AddAlarmGroupAction> = {
+        action: AddAlarmGroupAction,
+        reducer: (draftState, { alarmGroup }) => {
+            draftState.alarmGroups[alarmGroup.id] = alarmGroup;
+            return draftState;
+        },
+        rights: 'trainer',
+    };
+
+    export const renameAlarmGroup: ActionReducer<RenameAlarmGroupAction> = {
+        action: RenameAlarmGroupAction,
+        reducer: (draftState, { alarmGroupId, name }) => {
+            const alarmGroup = getElement(
+                draftState,
+                'alarmGroups',
+                alarmGroupId
+            );
+            alarmGroup.name = name;
+            return draftState;
+        },
+        rights: 'trainer',
+    };
+
+    export const removeAlarmGroup: ActionReducer<RemoveAlarmGroupAction> = {
+        action: RemoveAlarmGroupAction,
+        reducer: (draftState, { alarmGroupId }) => {
+            getElement(draftState, 'alarmGroups', alarmGroupId);
+            delete draftState.alarmGroups[alarmGroupId];
+            return draftState;
+        },
+        rights: 'trainer',
+    };
+
+    export const addAlarmGroupVehicle: ActionReducer<AddAlarmGroupVehicleAction> =
+        {
+            action: AddAlarmGroupVehicleAction,
+            reducer: (draftState, { alarmGroupId, alarmGroupVehicle }) => {
+                const alarmGroup = getElement(
+                    draftState,
+                    'alarmGroups',
+                    alarmGroupId
+                );
+                alarmGroup.alarmGroupVehicles[alarmGroupVehicle.id] =
+                    alarmGroupVehicle;
+                return draftState;
+            },
+            rights: 'trainer',
+        };
+
+    export const editAlarmGroupVehicle: ActionReducer<EditAlarmGroupVehicleAction> =
+        {
+            action: EditAlarmGroupVehicleAction,
+            reducer: (
+                draftState,
+                { alarmGroupId, alarmGroupVehicleId, time }
+            ) => {
+                const alarmGroup = getElement(
+                    draftState,
+                    'alarmGroups',
+                    alarmGroupId
+                );
+                if (!alarmGroup.alarmGroupVehicles[alarmGroupVehicleId]) {
+                    throw new ReducerError(
+                        `AlarmGroupVehicle with id ${alarmGroupVehicleId} does not exist in Alarmgroup with id ${alarmGroupId}`
+                    );
+                }
```

Comment:
Add a `getAlarmGroupVehicle(alarmGroupId, alarmGroupVehicleId)` function for this. (Similar to [this](https://github.com/hpi-sam/digital-fuesim-manv/blob/92f30c933ca001cb04fe964f8e56bd4c0672e05b/shared/src/store/action-reducers/map-image-templates.ts#L90) function)

Reply:
Similar in the sense of the position in the file